### PR TITLE
Switch to use golangci-lint

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,47 +1,27 @@
...
+    - goimports
+    - revive
     - misspell
-    - errcheck
-    - gosimple
-    - staticcheck
-    - gosec
-    - gocritic
-    - unparam
-    - deadcode
```
Comment (khanhtc1202) [inline .golangci.yml:14]: should we keep this? and the `exportloopref` as well 🤔

Comment (nghialv) [reply .golangci.yml:14]: Right. Let me revert them. (Just commented out them to check the action before.)

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -9,15 +9,18 @@ on:
...
   go:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Run lint
-        run: make lint/go
+      - name: golangci-lint
+        uses: reviewdog/action-golangci-lint@f3dc5fadcaff5d8da3574b129a58db433171b1a8 #v2.1.3
+        with:
+          go_version: ${{ env.GO_VERSION }}
```
Comment (knanao) [inline .github/workflows/lint.yaml:23]: [nits] Is it better we specify `golangci_lint_version` expressively ?

Comment (knanao) [reply .github/workflows/lint.yaml:23]: tbh, it's okay not to be set that one because we are using `disable-all`. it's up to you🙌

Comment (nghialv) [reply .github/workflows/lint.yaml:23]: You're right. Let me add it.

**Review by knanao — APPROVED** [review]
Thanks!

**Review by khanhtc1202 — APPROVED** [review]
Cool

**Pull request merged**